### PR TITLE
onboarding Done implementation and bug fix

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html [hunk 1]
```diff
@@ -8,10 +8,15 @@ <h5 class="tw-text-14-px">{{ label }}</h5>
...
     </div>
-    <div class="tw-flex tw-pl-40-px">
+    <div *ngIf="options" class="tw-flex tw-pl-40-px">
         <div *ngFor="let option of options" class="p-field-checkbox tw-mr-20-px">
             <p-radioButton class="tw-mr-10-px" [value]="option.value" [formControlName]="formControllerName"></p-radioButton>
-            <label>{{option.label}}</label>
+            <label class="!tw-text-14-px">{{option.label}}</label>
         </div>
     </div>
+    <div *ngIf="!options">
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html]: if we are not using radio button, then why can't we just display this in the parent component. Doesn't make sense to pass a label to child component.

File: src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.html [hunk 2]
```diff
@@ -10,35 +10,64 @@ <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px" [innerHTML]="subL
         </div>
     </div>
     <div>
-        <div class="p-field-checkbox tw-mr-20-px tw-pl-40-px" *ngIf="formControllerName !== 'dayOfMonth' && formControllerName !== 'dayOfWeek' && formControllerName !== 'timeOfDay'">
-            <p-dropdown [options]="options" class="!tw-w-300-px" [formControlName]="formControllerName" [placeholder]="placeholder"></p-dropdown>
+        <div *ngIf="!options && !simpleOptions">
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.html:13]: same comment as above

File: src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts [hunk 3]
```diff
@@ -215,8 +211,8 @@ export class ExportSettingComponent implements OnInit {
         reimbursableExportDate: [this.exportSettings?.reimbursable_expense_date ? this.exportSettings?.reimbursable_expense_date : null],
         creditCardExpense: [this.exportSettings?.credit_card_expense_export_type ? true : false, this.exportSelectionValidator()],
         cccExportType: [this.exportSettings?.credit_card_expense_export_type ? this.exportSettings?.credit_card_expense_export_type : null],
-        cccExportGroup: [this.exportSettings?.credit_card_expense_grouped_by ? this.exportSettings?.credit_card_expense_grouped_by : null],
-        cccExportDate: [this.exportSettings?.credit_card_expense_date ? this.exportSettings?.credit_card_expense_date : null],
+        cccExportGroup: [this.exportSettings?.credit_card_expense_grouped_by ? this.exportSettings?.credit_card_expense_grouped_by : this.expenseGroupingFieldOptions[1].value],
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts:220]: we shoudn't pick this way `this.expenseGroupingFieldOptions[1].value]`. Check QBO impl.

File: src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts [hunk 4]
```diff
@@ -215,8 +211,8 @@ export class ExportSettingComponent implements OnInit {
...
+        cccExportDate: [this.exportSettings?.credit_card_expense_date ? this.exportSettings?.credit_card_expense_date : this.reimbursableExpenseGroupingDateOptions[0].value],
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts:221]: same

File: src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts [hunk 5]
```diff
@@ -233,8 +229,8 @@ export class ExportSettingComponent implements OnInit {
           reimbursableExportDate: [null],
           creditCardExpense: [false, this.exportSelectionValidator()],
           cccExportType: [null],
-          cccExportGroup: [null],
-          cccExportDate: [null],
+          cccExportGroup: [this.expenseGroupingFieldOptions[1].value],
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts:238]: why are we defaulting values and not null?

File: src/app/shared/components/qbd/configuration/advanced-setting/advanced-setting.component.ts [hunk 6]
```diff
@@ -114,7 +116,16 @@ export class AdvancedSettingComponent implements OnInit {
...
+    const finnaltime = time-12;
+    if (time<=12) {
+      if (time>0 && time <=9) {
+        return "0" + time.toString() + ":" + seconds;
+      } else if (time === 0) {
+        return "12:" + seconds;
+      }
+      return time.toString() + ':' + seconds;
+    }
+    return finnaltime > 9 ? finnaltime + ":" + seconds : "0" + finnaltime.toString() + ":" + seconds;
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/advanced-setting/advanced-setting.component.ts:128]: in this section we are trying to convert the 24hrs time from the response to 12hrs time, 
if time is greater that 12, 
    we are subtracting the hrs we got with 12 > we get number less that or equal to 12,
           if the resultant is greater that 9 we form the time by concating the resultant hrs with seconds
           else we are adding a "0" before the resultant time and concating the resultant hrs with seconds

File: src/app/shared/components/qbd/configuration/advanced-setting/advanced-setting.component.ts [hunk 7]
```diff
@@ -171,11 +182,13 @@ export class AdvancedSettingComponent implements OnInit {
 
   private constructPayloadAndSave(): void {
     this.saveInProgress = true;
+    const topMemo = [this.advancedSettingsForm.value.topMemoStructure[0]];
     const currentTime = +this.advancedSettingsForm.controls.timeOfDay.value.slice(0, 2);
     const currentMins = this.advancedSettingsForm.controls.timeOfDay.value.slice(3, 5);
-    const time = this.advancedSettingsForm.value.meridiem === 'PM' ? `${currentTime+12}:${currentMins}:00`: currentTime === 12 ? `00:${currentMins}:00` : `${currentTime}:${currentMins}:00`;
-    this.advancedSettingsForm.controls.timeOfDay.patchValue(time);
+    const time = this.advancedSettingsForm.value.meridiem === 'PM' && currentTime !== 12 ? currentTime+12 + ":" + currentMins+":00" : currentTime === 12 ? "00:" + currentMins+":00" : currentTime + ":" + currentMins + ":00";
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/advanced-setting/advanced-setting.component.ts:188]: we converting the value we got as 12hrs to 24hrs, as this is the formate of time in api

File: src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html (hunk 1 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html]: this is because we need to show the label for the field

File: src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.html (hunk 2 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.html:13]: @ashwin1111 this is because we need to show the label for the field

File: src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts (hunk 3 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts:220]: @ashwin1111  we defaulting values because, if the user clicks credit card purchase while saving that we need value so these values are defaulted

File: src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts (hunk 4 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts:221]: we defaulting values because, if the user clicks credit card purchase while saving that we need value so these values are defaulted

File: src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts (hunk 5 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/export-setting/export-setting.component.ts:238]: we defaulting values because, if the user clicks credit card purchase while saving that we need value so these values are defaulted

File: src/app/core/models/qbd/qbd-configuration/advanced-setting.model.ts [hunk 8]
```diff
@@ -34,13 +34,25 @@ export type QBDAdvancedSettingsGet = {
 
 export class AdvancedSettingModel {
     static constructPayload(advancedSettingForm: FormGroup): QBDAdvancedSettingsPost {
+        const topMemo: string[] = [];
+        topMemo.push(advancedSettingForm.value.topMemoStructure);
+        const currentTime = +advancedSettingForm.controls.timeOfDay.value.slice(0, 2);
+        const currentMins = advancedSettingForm.controls.timeOfDay.value.slice(3, 5);
+        let time = '';
+        if (advancedSettingForm.value.meridiem === 'PM' && currentTime !== 12) {
```
Comment (ashwin1111) [inline src/app/core/models/qbd/qbd-configuration/advanced-setting.model.ts:42]: this logic would break for AM time. time will be '' in that case

File: src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html (hunk 1 above)
Comment (ashwin1111) [reply src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html]: it doesn't make sense to show just label in a radio component right? can't we do this in parent directly?

Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html]: no we can't

Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html]: we can, but it will be like repeating the code

Comment (ashwin1111) [reply src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html]: can we have something like configuration-label component then?

File: src/app/core/models/qbd/qbd-configuration/advanced-setting.model.ts [hunk 9]
```diff
@@ -34,13 +34,25 @@ export type QBDAdvancedSettingsGet = {
...
+            time = currentTime+12 + ":" + currentMins+":00";
+          } else if (currentTime === 12) {
+            time = "00:" + currentMins+":00";
+          } else {
+            time = currentTime + ":" + currentMins + ":00";
```
Comment (DhaaraniCIT) [inline src/app/core/models/qbd/qbd-configuration/advanced-setting.model.ts:47]: like this

File: src/app/core/models/qbd/qbd-configuration/advanced-setting.model.ts (hunk 8 above)
Comment (DhaaraniCIT) [reply src/app/core/models/qbd/qbd-configuration/advanced-setting.model.ts:42]: i have a else case where the time will be set
https://github.com/fylein/fyle-integrations-settings-app/pull/40/files#r1104311177

**Comment by DhaaraniCIT** [comment]
https://user-images.githubusercontent.com/55541808/218453002-28c90a53-97ea-4664-a6fe-6129f65d42bc.mov

File: src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html (hunk 1 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/configuration-radio-field/configuration-radio-field.component.html]: ok

File: src/app/shared/components/qbd/configuration/configuration-label/configuration-label.component.html [hunk 10]
```diff
@@ -0,0 +1,18 @@
+<div [formGroup]="form" class="tw-p-24-px">
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-label/configuration-label.component.html:1]: why is form group required?

File: src/app/shared/components/qbd/configuration/configuration-label/configuration-label.component.html [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+                <svg-icon-sprite *ngIf="iconPath" src="{{iconPath}}" width="20px" hight="20px" class="tw-text-mandatory-field-color"></svg-icon-sprite>
+            </div>
+            <div class="tw-font-500 tw-leading-normal">
+                <h5 class="tw-text-14-px">{{ label }}<app-mandatory-field *ngIf="isFieldMandatory"></app-mandatory-field></h5>
+                <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px" [innerHTML]="subLabel"></h5>
+            </div>
+        </div>
+        <div class="tw-flex tw-pl-40-px">
+            <h4>
+                {{ expensesGroup.label }}
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-label/configuration-label.component.html:14]: name it in a generic way. It be just label. You can send the label directly from parent component. Value is not required since we are just displaying the value

File: src/app/shared/components/qbd/configuration/configuration-label/configuration-label.component.ts [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+@Component({
+  selector: 'app-configuration-label',
+  templateUrl: './configuration-label.component.html',
+  styleUrls: ['./configuration-label.component.scss']
+})
+export class ConfigurationLabelComponent implements OnInit {
+
+  @Input() expensesGroup: QBDExportSettingFormOption;
+
+  @Input() form: FormGroup;
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-label/configuration-label.component.ts:14]: why is this required?

File: src/app/shared/components/qbd/configuration/configuration-label/configuration-label.component.ts [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+  @Input() iconPath: string;
+
+  @Input() label: string;
+
+  @Input() subLabel: string;
+
+  @Input() placeholder: string;
+
+  @Input() formControllerName: string;
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-label/configuration-label.component.ts:24]: same

File: src/app/integrations/qbd/onboarding/onboarding-done/onboarding-done.component.html [hunk 14]
```diff
@@ -1 +1,21 @@
-<p>onboarding-done works!</p>
+<div class="tw-absolute tw-bg-header-1 tw-h-110-px tw-w-[100%] tw-z-[-1]"></div>
+<div class="tw-absolute tw-bg-header-2 tw-h-110-px tw-w-[100%] tw-z-[-1]"></div>
+<div class="tw-flex tw-justify-center tw-pr-74-px tw-pl-74-px tw-pt-24 tw-pb-20">
+  <div class="onboarding-done--contents">
+    <div class="tw-pt-34-px">
+      <img src="assets/icons/success-check.gif" class="onboarding-done--align-center" alt="Onboarding done" width="120px"
```
Comment (ashwin1111) [inline src/app/integrations/qbd/onboarding/onboarding-done/onboarding-done.component.html:6]: Screenshot 2023-02-13 at 7 23 55 PM
this is not taken care

**Review by ashwin1111 — APPROVED** [review]
